CMakeLists.txt

```diff
@@ -1,114 +1,85 @@
-# Setting up CMake Project
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.27)
 project(cartogram LANGUAGES CXX)
 
-# Setting compile options.
+# ========== Project Setup ==========
 set(CMAKE_CXX_STANDARD 20)
 set(THREADS_PREFER_PTHREAD_FLAG ON)
-
-# Setting CMake build options, adding path for binary.
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
-# Release mode by default, unless specified
-# https://stackoverflow.com/questions/48832233/
-# have-a-cmake-project-default-to-the-release-build-type
-if (NOT CMAKE_BUILD_TYPE)
+# Default build type
+if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Release" CACHE STRING "" FORCE)
 endif()
 
-# Lots of warnings.
-add_compile_options(-Wall -Wextra -pedantic)
+# ========== Dependencies Setup ==========
 
-# Finding appropriate packages.
-find_package(OpenMP REQUIRED)
-find_package(PkgConfig REQUIRED)
+# Boost
+find_package(Boost REQUIRED)
 
-# Finding and linking FFTW, and Cairo using PkgConfig.
-pkg_search_module(FFTW REQUIRED fftw3 IMPORTED_TARGET)
-pkg_search_module(Cairo REQUIRED cairo IMPORTED_TARGET)
-include_directories(PkgConfig::FFTW PkgConfig::Cairo)
-link_libraries(PkgConfig::FFTW PkgConfig::Cairo)
-
-
-# Adding C++ files needed for compilation.
-add_executable(
-  cartogram
-  src/main.cpp
-  src/cartogram_info/cartogram_info.cpp
-  src/cartogram_info/read_csv.cpp
-  src/cartogram_info/read_geojson.cpp
-  src/cartogram_info/shift_insets_to_position.cpp
-  src/cartogram_info/write_geojson.cpp
-  src/geo_div/geo_div.cpp
-  src/inset_state/albers_projection.cpp
-  src/inset_state/auto_color.cpp
-  src/inset_state/blur_density.cpp
-  src/inset_state/check_topology.cpp
-  src/inset_state/densify.cpp
-  src/inset_state/fill_with_density.cpp
-  src/inset_state/flatten_density.cpp
-  src/inset_state/inset_state.cpp
-  src/inset_state/interpolate_bilinearly.cpp
-  src/inset_state/matrix.cpp
-  src/inset_state/project.cpp
-  src/inset_state/rescale_map.cpp
-  src/inset_state/round_point.cpp
-  src/inset_state/scanline_graph.cpp
-  src/inset_state/simplify_inset.cpp
-  src/inset_state/smyth_projection.cpp
-  src/inset_state/write_cairo.cpp
-  src/inset_state/write_eps.cpp
-  src/inset_state/write_inset_to_geojson.cpp
-  src/misc/colors.cpp
-  src/misc/ft_real_2d.cpp
-  src/misc/intersection.cpp
-  src/misc/parse_arguments.cpp
-  src/misc/pwh.cpp
-)
+# Matplot++
+find_package(Matplot++ REQUIRED)
+
+# PkgConfig, fftw, and cairo
+find_package(PkgConfig REQUIRED)
+pkg_search_module(fftw REQUIRED fftw3 IMPORTED_TARGET)
+pkg_search_module(CAIRO REQUIRED CAIRO IMPORTED_TARGET)
 
-# Linking appropriate libraries required.
+# ========== Compiler Setup ==========
 if(APPLE)
+  set(LLVM_BASE_PATH "/usr/local/opt/llvm@13/bin/")
```

Comment:
After reading <https://releases.llvm.org/14.0.0/tools/clang/docs/ReleaseNotes.html#floating-point-support-in-clang> and running a test on Belgium, it appears to be possible to replace `llvm@13` with `llvm` and use  the switch `-ffp-contract=off` below.

Reply:
Latest `llvm@17` is used to make sure the compilation does not fail in case of untested inclusion of newer llvm version. `llvm@17` is tested and works as intended.

I have noticed that with `-ffp-contract=off`, the cartogram generation is bit slower. It is possible that for `llvm@13`, `-ffp-contract` was set to `on` by default.